google/cloud/spanner_v1/session.py

```diff
@@ -421,21 +421,13 @@ def run_in_transaction(self, func, *args, **kw):
         )
         attempts = 0
 
-        while True:
```

Comment:
I don't quite understand the need for refactoring here. Can you help me understand?
Just adding a trace_call with `CloudSpanner.run_in_transaction` before `while True` should work right?

Reply:
Gotcha! I really was just being very sensitive with it to ensure that if there were multiple invocations in that while loop of creating a new Transaction that we'd show them all but even better I have added span events for aborting, retrying et al